File: GooglePay/src/main/java/com/braintreepayments/api/GooglePayPaymentAuthRequest.java
package com.braintreepayments.api;

import androidx.annotation.NonNull;

import com.google.android.gms.wallet.PaymentDataRequest;

/**
 * Used to request Google Pay payment authorization via
 * {@link GooglePayLauncher#launch(GooglePayPaymentAuthRequest)}
 */
public class GooglePayPaymentAuthRequest {

    private final int googlePayEnvironment;
    private final PaymentDataRequest paymentDataRequest;

    GooglePayPaymentAuthRequest(int googlePayEnvironment, @NonNull PaymentDataRequest paymentDataRequest) {
        this.googlePayEnvironment = googlePayEnvironment;
        this.paymentDataRequest = paymentDataRequest;
    }

    int getGooglePayEnvironment() {
        return googlePayEnvironment;
    }

    PaymentDataRequest getPaymentDataRequest() {
        return paymentDataRequest;
    }
}

File: GooglePay/src/main/java/com/braintreepayments/api/GooglePayPaymentAuthRequest.kt
package com.braintreepayments.api

/**
 * A request used to launch the Venmo app for continuation of the Google Pay flow.
 */
sealed class GooglePayPaymentAuthRequest {

    /**
     * The request was successfully created and is ready to be launched by [GooglePayLauncher]
     */
    class ReadyToLaunch(val requestParams: GooglePayPaymentAuthRequestParams) : GooglePayPaymentAuthRequest()

    /**
     * There was an [error] creating the request
     */
    class Failure(val error: Exception) : GooglePayPaymentAuthRequest()
}
